### PR TITLE
REUPLOAD Added vanilla LSTM and LSTM with peepholes oneDNN fp32 kernel

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
This PR adds vanilla LSTM and LSTM with peepholes oneDNN kernel to PaddlePaddle. Currently only fp32 is supported, but bf16 and int8 will be added in future.

### CODE REVIEW

**Comment by jakpiase** [comment]
@jczaja I have done minor changes since last PR, could you please take a look?

File: paddle/fluid/operators/fused/mkldnn/fusion_lstm_mkldnn_op.cc [hunk 1]
```diff
@@ -0,0 +1,377 @@
...
+    if (use_peepholes) {
+      auto peephole_weight_p = handler.AcquirePeepholeWeights(bias);
+      std::pair<int, dnnl::memory> peepholes_weights(DNNL_ARG_WEIGHTS_PEEPHOLE,
+                                                     *peephole_weight_p);
+      lstm_args.insert(peepholes_weights);
+    }
+
+    auto lstm_forward_p = handler.AcquireForwardPrimitive();
+
+    dnnl::stream astream(mkldnn_engine);
```
Comment (jczaja) [inline paddle/fluid/operators/fused/mkldnn/fusion_lstm_mkldnn_op.cc]: Please read stream from TLS , avoid creatgin new one

File: paddle/fluid/operators/fused/mkldnn/fusion_rnn_mkldnn.h [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+        user_h0_memory = dnnl::memory({{1, 1, N, OC},
+                                       MKLDNNGetDataType<float>(),
+                                       MKLDNNMemoryFormat::ldnc},
+                                      this->engine_);
+        memset(user_h0_memory.get_data_handle(), 0, sizeof(float) * N * OC);
+      }
+      memory_p = std::make_shared<dnnl::memory>(this->fwd_pd_->src_iter_desc(),
+                                                this->engine_);
+
+      dnnl::stream astream(this->engine_);
```
Comment (jczaja) [inline paddle/fluid/operators/fused/mkldnn/fusion_rnn_mkldnn.h]: read stream from TLS

**Review by jczaja — APPROVED** [review]
LGTM

**Comment by jczaja** [comment]
@luotao1 This is resolved version of LSTM fp32 oneDNN PR that was reverted. Please review.

**Review by luotao1 — APPROVED** [review]
LGTM

**Pull request merged**